src/org/labkey/test/util/TestDataGenerator.java

```diff
@@ -304,7 +304,8 @@ public String writeTsvContents()
         {
```

Comment:
The problem here was that the key here is the name of the column, not its ordinal.  Thus, calling `row.get(_indices.get(index).getName())` returned null because there were no numeric keys.  I suspect this was broken when we updated the structure of _indices a long time ago.

One potentially problematic issue with this solution is that it toLowerCases `keyAtIndex` and that might not be right(but it was the only way I could make this work).  
The key of the row is the column name, in this case if I passed the name pulled out of the PropertyDescriptor and it missed because the values in `row `were all lower case, the names were camel-cased.  I've had success in the past resolving case-insensitive key mismatches by streaming over a set to find the one that matches to-lower and then passing that as the key, but I suspect there's a better long-term fix to this than that.

Reply:
You could use `CaseInsensitiveHashMap` (from our remoteapi) for the rows. It exists for precisely this sort of thing.

Reply:
If I make rows into a List<CaseInsensitiveHashMap>, I run into problems trying to pass rows into an insertRowsCommand- it expects a List<Map<String, Object>>

Reply:
I figured out why the case was mismatched- when generating rows with random data, I was creating them with column name and type toLowerCase(), which caused the mismatch here.  By no longer doing that (and by instead switching over columnType.toLowerCase when setting dataSuppliers for columns to ensure a match) I think I've got a fix.

I'd like to discuss using a CaseInsensitiveHashMap here, but I worry that it will require changing a lot of test code (and our remoteAPI wrappers, perhaps)

Reply:
I don't think `columnType.toLowerCase` will make any difference here. The mismatch is with the name, not the type, right?
As for using CIHashMap; it wouldn't require much refactor. I think the only change needed would be to update `addCustomRow` and `generateRows` to use it. (See #686)